src/pcapng.h

```diff
@@ -68,6 +73,8 @@ struct PCAPNG_BLOCK_TAIL {
 };
 #include <poppack.h>
 
+struct PCAPNG_BLOCK_OPTION_ENDOFOPT EndOption = { .Code = PCAPNG_OPTIONCODE_ENDOFOPT, .Length = 0};
```

Comment:
Don't make this global.

Reply:
why?

Reply:
All you've done is remove like three lines of duplicated code. This did not need to be refactored.

Reply:
It's much cleaner to me. We don't need to set code and set length in each function.

Reply:
More importantly, it's less error-prone because you don't need to do that in each function. And btw, it's removing 6 lines of code. I'd like to leave it as is.